windows/RNCNetInfoCPP/RNCNetInfo.h

```diff
@@ -15,10 +15,16 @@ namespace winrt::ReactNativeNetInfo::implementation {
     struct NetInfoDetails {
         REACT_FIELD(isConnectionExpensive);
         bool isConnectionExpensive;
+    };
 
+    REACT_STRUCT(NetInfoCellularDetails);
+    struct NetInfoCellularDetails : NetInfoDetails {

```

Comment:
I don't think inheritance will work here.

The NetInfoState.details field is storing a statically allocated NetInfoDetails object, which only has a single boolean field. You can't just inherit from that class, add a bunch of fields, and expect that your new expanded object can fit in the same memory space as the base class instance. When you assign a derived NetInfoCellularDetails object to the details field, only isConnectionExpensive get copied over and when the details field is serialized only that one field will be written out.

Attempts to write a NetInfoCellularDetails into the details field will

Reply:
My issue is that it breaks the contract stated in the README so the onus is on the user to filter out those values but you're right about the REACT_FIELD not properly handling pointers